approvals/approvals/api.py

```diff
@@ -54,8 +62,8 @@ def get_document_approvals(doc: Document, method: str | None = None):
 
 
 @frappe.whitelist()
-def fetch_approvals_and_roles(doc: Document, method: str | None = None):
-	doc = frappe._dict(json.loads(doc)) if isinstance(doc, str) else doc
+def fetch_approvals_and_roles(doc: Document | str, method: str | None = None):
+	doc = frappe.get_doc(json.loads(doc)) if isinstance(doc, str) else doc
```

Comment:
@agritheory I've changed the `frappe._dict` calls to `get_doc` calls to keep typing consistency. This might add a slight overhead, but it'll let us use document APIs like `add_comment` directly.